src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -566,30 +583,65 @@ export class ResultsViewPageStore
         this.driverAnnotationSettings = buildDriverAnnotationSettings(
             () => store.didHotspotFailInOncoprint
         );
-        this.driverAnnotationsReactionDisposer = reaction(
-            () => this.urlWrapper.query.cancer_study_list,
-            () => {
-                this.driverAnnotationSettings = buildDriverAnnotationSettings(
-                    () => store.didHotspotFailInOncoprint
-                );
-            },
-            { fireImmediately: true }
+
+        this.pageUserSession = new PageUserSession<ResultPageSettings>(
+            appStore,
+            ServerConfigHelpers.sessionServiceIsEnabled()
+        );
+
+        this.pageUserSession.id = {
+            page: PageType.RESULTS_VIEW,
+            origin: this.cancerStudyIds,
+        };
+
+        this.reactionDisposers.push(
+            reaction(
+                () => this.urlWrapper.query.cancer_study_list,
+                () => {
+                    this.driverAnnotationSettings = buildDriverAnnotationSettings(
+                        () => store.didHotspotFailInOncoprint
+                    );
+                },
+                { fireImmediately: true }
+            )
+        );
+
+        this.reactionDisposers.push(
+            reaction(
+                () => [this.cancerStudyIds],
+                () => {
+                    this.pageUserSession.id = {
+                        page: PageType.RESULTS_VIEW,
+                        origin: this.cancerStudyIds,
+                    };
+                }
+            )
+        );
+
+        this.reactionDisposers.push(
```

Comment:
if we aren't tracking clinical tracks in url, why do we need this?

Reply:
Fixed in [d44a497](https://github.com/cBioPortal/cbioportal-frontend/compare/9065f7ea67dda257aad80c7a2b09cc88272492d9..d44a497fd7eb858dfe038ae4f6050daad90b67d9)